networking_aci/extensions/acioperations.py

```diff
@@ -0,0 +1,219 @@
+extensions.register_custom_supported_check(Acioperations.get_alias(), lambda: True, True)
+extensions.append_api_extensions_path(networking_aci.extensions.__path__)
+
+
+class ConfigController(wsgi.Controller):
+    def __init__(self, plugin, db):
+        super(ConfigController, self).__init__()
+        self.plugin = plugin
+
+    def index(self, request, **kwargs):
+        """Show all hostgroup, fixed bindings and address scopes"""
+        check_cloud_admin(request)
+        if request.GET.get("raw", False):
+            hostgroups = ACI_CONFIG.hostgroups
+        else:
+            hostgroups = {}
+            for hg_name in ACI_CONFIG.hostgroups:
+                hostgroups[hg_name] = ACI_CONFIG.get_hostgroup(hg_name)
+
+        return {
+            'hostgroups': hostgroups,
+            'fixed_bindings': ACI_CONFIG.fixed_bindings,
+            'address_scopes': ACI_CONFIG.address_scopes,
+        }
+        # return {"status": "success", "index": True, "arguments": kwargs}
+
+    def show(self, request, **kwargs):
+        """Show a single hostgroup"""
+        check_cloud_admin(request)
+        hostgroup_name = kwargs.pop('id')
+        if request.GET.get("raw", False):
+            hostgroup = ACI_CONFIG.hostgroups.get(hostgroup_name)
+        else:
+            hostgroup = ACI_CONFIG.get_hostgroup(hostgroup_name)
+
+        if not hostgroup:
+            raise web_exc.HTTPNotFound('Hostgroup "{}" does not exist'.format(hostgroup_name))
+
+        return hostgroup
+
+
+class HostgroupModeController(wsgi.Controller):
+    def __init__(self, plugin, db, rpc_notifier):
+        super(HostgroupModeController, self).__init__()
+        self.plugin = plugin
+        self.db = db
+        self.rpc_notifier = rpc_notifier
+
+    def index(self, request, **kwargs):
+        check_cloud_admin(request)
+        ctx = context.get_admin_context()
+        return dict(hostgroup_modes=self.db.get_hostgroup_modes(ctx))
+
+    def show(self, request, **kwargs):
+        check_cloud_admin(request)
+        hostgroup_name = kwargs.pop('id')
+        ctx = context.get_admin_context()
+        return dict(hostgroup_name=hostgroup_name, hostgroup_mode=self._get_mode(ctx, hostgroup_name))
+
+    def update(self, request, **kwargs):
+        check_cloud_admin(request)
+        hostgroup_name = kwargs.pop('id')
+        new_mode = kwargs.get("body", {}).get("mode")
+        if new_mode not in (aci_const.MODE_INFRA, aci_const.MODE_BAREMETAL):
+            raise web_exc.HTTPBadRequest("Specify a mode that is either {} or {} (given mode was '{}')"
+                                         .format(aci_const.MODE_INFRA, aci_const.MODE_BAREMETAL, new_mode))
+
+        ctx = context.get_admin_context()
+        curr_mode = self._get_mode(ctx, hostgroup_name)
+        if curr_mode == new_mode:
+            raise web_exc.HTTPBadRequest("{} is already in {} mode".format(hostgroup_name, new_mode))
+
+        # check if there are still  on this hostgroup
```

Comment:
I think there's a word missing in this comment (is it "segment" or "portbinding"?) - or a space too much.